lib/resty/jwt.lua

```diff
@@ -452,28 +466,51 @@ end
 
 --@function validate_exp_nbf - validate expiry and not valid before
 --@param jwt_obj
-local function validate_exp_nbf(jwt_obj, leeway)
+local function validate_exp_nbf(jwt_obj, is_validation_loose, leeway)
+  if jwt_obj[str_const.reason] ~= nil then
+    return
+  end
+
   local exp = jwt_obj[str_const.payload][str_const.exp]
   local nbf = jwt_obj[str_const.payload][str_const.nbf]
 
-  if (exp ~= nil or nbf ~= nil ) and not jwt_obj[str_const.reason] then
-    leeway = leeway or 0
-    local now = ngx.now()
+  leeway = leeway or 0
+  local now = ngx.now()
+
+  if exp ~= nil and not is_validation_loose then
```

Comment:
if `is_validation_loose` is `true`.
`exp` checking is disabled.

but what we really want to do

present | valid | loose | result
------- | ------- | ------- | -------
√ | √ | - | perform checking
√ | x | √ | skip the checking
√ | x | x | raise 500 with meaningful message
x | - | - | skip the checking

is that right :question: